### DISCUSSION TITLE
[urgent] Unable to migrate

### DISCUSSION THREAD

**Discussion by vasujogani** [prisma/prisma#20368] [Q&A]
### Bug description

I have a local postgres db running on docker.

Everytime i restart the instance and rebuild the container, prisma clients fails to match the changes.

Here is `prisma migrate status` response

```
4 migrations found in prisma/migrations

Your local migration history and the migrations table from your database are different:

The last common migration is: null

The migrations have not yet been applied:
20230702022338_A
20230702050144_B
20230703014427_C
20230703054338_D

The migration from the database are not found locally in prisma/migrations:
init

```

It gives an error

```
✔ We need to reset the "public" schema at "127.0.0.1:6500"
Do you want to continue? All data will be lost. … no

```

### How to reproduce

1. Migate something via prisma on a postgres db that is running on docker
2. delete the postgres container and restart (make sure its linked to local file system so restarted instance is same as before)
3. add a new field to prisma.schema file and try re-migrating

### Expected behavior

It should pick up the difference between the prisma schema and postgres and only apply that
*I dont want to delete all my data*

### Prisma information

```  
 "@prisma/client": "^4.16.2",
```

### Environment & setup

- Node.js version: 18.12.1

### Prisma Version

```
4.16.2
```

**Comment by nurul3101** [comment]
Hello @vasujogani 👋 

Prisma uses a specific table called `_prisma_migrations` in your database to keep track of which migrations have been applied.

If you delete your PostgreSQL container, this table will also be deleted, and Prisma will lose track of the applied migrations. Therefore, when you restart the container and try to apply migrations again, Prisma will not recognise the existing database schema and will try to apply all migrations from scratch.

Is it necessary for your use case to delete and recreate the container? Instead of deleting and recreating the container can you stop and start the container as needed?

**Reply by vasujogani** [reply]
Sometimes we have to delete it. But it is linked to my local file system, should it not also store `_prisma_migrations`? 

When I recreate my postgres db, it has all the tables except `_prisma_migrations`

**Reply by nurul3101** [reply]
How are you recreating the Postgres DB?
Are you creating it from a backup?

**Reply by nurul3101** [reply]
Checking to see if you had a chance to look at my last response?

**Reply by vasujogani** [reply]
Yeah, the volumes are shared so it creates the tables when the container is recreated

**Reply by nurul3101** [reply]
Would it be possible for you to provide a minimal reproduction?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
